jetty-util/src/main/java/org/eclipse/jetty/util/MultiPartInputStreamParser.java

```diff
@@ -642,11 +655,11 @@ else if(tl.startsWith("filename="))
                 InputStream partInput = null;
                 if ("base64".equalsIgnoreCase(contentTransferEncoding))
                 {
-                    partInput = new Base64InputStream((ReadLineInputStream)_in);
+                    partInput = new Base64InputStream(is);
                 }
                 else if ("quoted-printable".equalsIgnoreCase(contentTransferEncoding))
                 {
-                    partInput = new FilterInputStream(_in)
+                    partInput = new FilterInputStream(is)
```

Comment:
Would it not be better to use the JVMs BufferedInputStream as the basis rather than doing your own buffering?  The buf filed is protected in that class and thus available to be used.